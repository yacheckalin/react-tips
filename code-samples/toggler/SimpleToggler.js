import React, { useState } from "react";
import PropTypes from "prop-types";

const SimpleToggler = ({ value } = { value: false }) => {
  const [toggle, setToggle] = useState(value);
  return (
    <div className="container">
      <div className="switch">
        <label>
          Off
          <input
            type="checkbox"
            onChange={(e) => setToggle(!toggle)}
            value={toggle}
          />
          <span className="lever"></span>
          On
        </label>
      </div>
    </div>
  );
};

SimpleToggler.propTypes = {
  value: PropTypes.bool.isRequired,
};

export default SimpleToggler;
