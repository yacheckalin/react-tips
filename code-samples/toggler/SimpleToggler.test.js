import React from "react";
import toJSON from "enzyme-to-json";
import { shallow } from "enzyme";

import SimpleToggler from "./SimpleToggler";

describe("<SimpleToggler />", () => {
  test("displays and renders properly", () => {
    const wrapper = shallow(<SimpleToggler value={false} />);

    expect(toJSON(wrapper)).toMatchSnapshot();
  });

  test("initialily renders off", () => {
    const wrapper = shallow(<SimpleToggler value={false} />);

    expect(wrapper.find("input").props().value).toBe(false);
  });
});
