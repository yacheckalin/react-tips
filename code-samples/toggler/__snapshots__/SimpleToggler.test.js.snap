// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SimpleToggler /> displays and renders properly 1`] = `
<div
  className="container"
>
  <div
    className="switch"
  >
    <label>
      Off
      <input
        onChange={[Function]}
        type="checkbox"
        value={false}
      />
      <span
        className="lever"
      />
      On
    </label>
  </div>
</div>
`;
